From 2601084299c745a857c47787330388136a99f33d Mon Sep 17 00:00:00 2001
From: Martin Kuhn <martin@kuhn.host>
Date: Sat, 31 Jan 2026 14:26:37 +0100
Subject: [PATCH] Patched for Raspberry Pi 5

Apply kernel config changes for RPi5 support on upstream kernel:
- Switch to 16K pages for arm64
- Build critical drivers as builtin (network, storage, HID, virtio)
- Enable IMA (Integrity Measurement Architecture)
- Enable PHY timestamping, SPI/PWM GPIO
- Set ondemand as default CPU frequency governor
- Disable ZSWAP, DRM_PANTHOR
---
 kernel/build/config-arm64 | 283 +++++++++++++++++++-------------------
 1 file changed, 140 insertions(+), 143 deletions(-)

diff --git a/kernel/build/config-arm64 b/kernel/build/config-arm64
index 399028b..a5539f4 100644
--- a/kernel/build/config-arm64
+++ b/kernel/build/config-arm64
@@ -460,8 +460,8 @@ CONFIG_ROCKCHIP_ERRATUM_3588001=y
 CONFIG_SOCIONEXT_SYNQUACER_PREITS=y
 # end of ARM errata workarounds via the alternatives framework
 
-CONFIG_ARM64_4K_PAGES=y
-# CONFIG_ARM64_16K_PAGES is not set
+# CONFIG_ARM64_4K_PAGES is not set
+CONFIG_ARM64_16K_PAGES=y
 # CONFIG_ARM64_64K_PAGES is not set
 # CONFIG_ARM64_VA_BITS_39 is not set
 CONFIG_ARM64_VA_BITS_48=y
@@ -666,9 +666,9 @@ CONFIG_CPU_FREQ_STAT=y
 # CONFIG_CPU_FREQ_DEFAULT_GOV_PERFORMANCE is not set
 # CONFIG_CPU_FREQ_DEFAULT_GOV_POWERSAVE is not set
 # CONFIG_CPU_FREQ_DEFAULT_GOV_USERSPACE is not set
-# CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND is not set
+CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND=y
 # CONFIG_CPU_FREQ_DEFAULT_GOV_CONSERVATIVE is not set
-CONFIG_CPU_FREQ_DEFAULT_GOV_SCHEDUTIL=y
+# CONFIG_CPU_FREQ_DEFAULT_GOV_SCHEDUTIL is not set
 CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
 CONFIG_CPU_FREQ_GOV_POWERSAVE=y
 CONFIG_CPU_FREQ_GOV_USERSPACE=y
@@ -974,7 +974,7 @@ CONFIG_BLK_DEV_WRITE_MOUNTED=y
 # CONFIG_BLK_DEV_ZONED is not set
 CONFIG_BLK_DEV_THROTTLING=y
 # CONFIG_BLK_WBT is not set
-CONFIG_BLK_CGROUP_IOLATENCY=y
+# CONFIG_BLK_CGROUP_IOLATENCY is not set
 # CONFIG_BLK_CGROUP_FC_APPID is not set
 # CONFIG_BLK_CGROUP_IOCOST is not set
 # CONFIG_BLK_CGROUP_IOPRIO is not set
@@ -1108,28 +1108,7 @@ CONFIG_COREDUMP=y
 # Memory Management options
 #
 CONFIG_SWAP=y
-CONFIG_ZSWAP=y
-# CONFIG_ZSWAP_DEFAULT_ON is not set
-# CONFIG_ZSWAP_SHRINKER_DEFAULT_ON is not set
-# CONFIG_ZSWAP_COMPRESSOR_DEFAULT_DEFLATE is not set
-# CONFIG_ZSWAP_COMPRESSOR_DEFAULT_LZO is not set
-# CONFIG_ZSWAP_COMPRESSOR_DEFAULT_842 is not set
-# CONFIG_ZSWAP_COMPRESSOR_DEFAULT_LZ4 is not set
-# CONFIG_ZSWAP_COMPRESSOR_DEFAULT_LZ4HC is not set
-CONFIG_ZSWAP_COMPRESSOR_DEFAULT_ZSTD=y
-CONFIG_ZSWAP_COMPRESSOR_DEFAULT="zstd"
-CONFIG_ZSMALLOC=y
-
-#
-# Zsmalloc allocator options
-#
-
-#
-# Zsmalloc is a common backend allocator for zswap & zram
-#
-# CONFIG_ZSMALLOC_STAT is not set
-CONFIG_ZSMALLOC_CHAIN_SIZE=8
-# end of Zsmalloc allocator options
+# CONFIG_ZSWAP is not set
 
 #
 # Slab allocator options
@@ -1220,7 +1199,6 @@ CONFIG_ARCH_HAS_ZONE_DMA_SET=y
 CONFIG_ZONE_DMA=y
 CONFIG_ZONE_DMA32=y
 CONFIG_ZONE_DEVICE=y
-CONFIG_HMM_MIRROR=y
 CONFIG_GET_FREE_REGION=y
 CONFIG_DEVICE_PRIVATE=y
 CONFIG_ARCH_USES_HIGH_VMA_FLAGS=y
@@ -1397,7 +1375,7 @@ CONFIG_NETLABEL=y
 # CONFIG_MPTCP is not set
 CONFIG_NETWORK_SECMARK=y
 CONFIG_NET_PTP_CLASSIFY=y
-# CONFIG_NETWORK_PHY_TIMESTAMPING is not set
+CONFIG_NETWORK_PHY_TIMESTAMPING=y
 CONFIG_NETFILTER=y
 CONFIG_NETFILTER_ADVANCED=y
 CONFIG_BRIDGE_NETFILTER=y
@@ -1460,7 +1438,7 @@ CONFIG_NF_TABLES_NETDEV=y
 CONFIG_NFT_NUMGEN=y
 CONFIG_NFT_CT=y
 # CONFIG_NFT_EXTHDR_DCCP is not set
-CONFIG_NFT_CONNLIMIT=y
+# CONFIG_NFT_CONNLIMIT is not set
 CONFIG_NFT_LOG=y
 CONFIG_NFT_LIMIT=y
 CONFIG_NFT_MASQ=y
@@ -2473,7 +2451,7 @@ CONFIG_BLK_DEV_UBLK=m
 #
 CONFIG_NVME_AUTH=m
 CONFIG_NVME_CORE=y
-CONFIG_BLK_DEV_NVME=m
+CONFIG_BLK_DEV_NVME=y
 CONFIG_NVME_MULTIPATH=y
 # CONFIG_NVME_VERBOSE_ERRORS is not set
 CONFIG_NVME_HWMON=y
@@ -2508,7 +2486,7 @@ CONFIG_NVME_TARGET_AUTH=y
 # CONFIG_ICS932S401 is not set
 CONFIG_ENCLOSURE_SERVICES=y
 # CONFIG_HI6421V600_IRQ is not set
-CONFIG_HP_ILO=m
+CONFIG_HP_ILO=y
 # CONFIG_QCOM_COINCELL is not set
 # CONFIG_QCOM_FASTRPC is not set
 # CONFIG_APDS9802ALS is not set
@@ -2610,7 +2588,7 @@ CONFIG_ISCSI_BOOT_SYSFS=y
 # CONFIG_SCSI_BNX2X_FCOE is not set
 # CONFIG_BE2ISCSI is not set
 # CONFIG_BLK_DEV_3W_XXXX_RAID is not set
-CONFIG_SCSI_HPSA=m
+CONFIG_SCSI_HPSA=y
 # CONFIG_SCSI_3W_9XXX is not set
 # CONFIG_SCSI_3W_SAS is not set
 # CONFIG_SCSI_ACARD is not set
@@ -2628,13 +2606,13 @@ CONFIG_SCSI_HISI_SAS_PCI=m
 # CONFIG_SCSI_ESAS2R is not set
 # CONFIG_MEGARAID_NEWGEN is not set
 # CONFIG_MEGARAID_LEGACY is not set
-CONFIG_MEGARAID_SAS=m
-CONFIG_SCSI_MPT3SAS=m
+CONFIG_MEGARAID_SAS=y
+CONFIG_SCSI_MPT3SAS=y
 CONFIG_SCSI_MPT2SAS_MAX_SGE=128
 CONFIG_SCSI_MPT3SAS_MAX_SGE=128
 # CONFIG_SCSI_MPT2SAS is not set
 CONFIG_SCSI_MPI3MR=m
-CONFIG_SCSI_SMARTPQI=m
+CONFIG_SCSI_SMARTPQI=y
 # CONFIG_SCSI_HPTIOP is not set
 # CONFIG_SCSI_BUSLOGIC is not set
 # CONFIG_SCSI_MYRB is not set
@@ -2684,7 +2662,7 @@ CONFIG_SATA_PMP=y
 #
 # Controllers with non-SFF native interface
 #
-CONFIG_SATA_AHCI=m
+CONFIG_SATA_AHCI=y
 CONFIG_SATA_MOBILE_LPM_POLICY=0
 CONFIG_SATA_AHCI_PLATFORM=y
 # CONFIG_AHCI_BRCM is not set
@@ -2731,7 +2709,7 @@ CONFIG_SATA_RCAR=y
 # PATA SFF controllers with BMDMA
 #
 # CONFIG_PATA_ALI is not set
-CONFIG_PATA_AMD=m
+CONFIG_PATA_AMD=y
 # CONFIG_PATA_ARTOP is not set
 # CONFIG_PATA_ATIIXP is not set
 # CONFIG_PATA_ATP867X is not set
@@ -2746,17 +2724,17 @@ CONFIG_PATA_AMD=m
 # CONFIG_PATA_IT8213 is not set
 # CONFIG_PATA_IT821X is not set
 # CONFIG_PATA_JMICRON is not set
-CONFIG_PATA_MARVELL=m
+CONFIG_PATA_MARVELL=y
 # CONFIG_PATA_NETCELL is not set
 # CONFIG_PATA_NINJA32 is not set
 # CONFIG_PATA_NS87415 is not set
-CONFIG_PATA_OLDPIIX=m
+CONFIG_PATA_OLDPIIX=y
 # CONFIG_PATA_OPTIDMA is not set
 # CONFIG_PATA_PDC2027X is not set
 # CONFIG_PATA_PDC_OLD is not set
 # CONFIG_PATA_RADISYS is not set
 # CONFIG_PATA_RDC is not set
-CONFIG_PATA_SCH=m
+CONFIG_PATA_SCH=y
 # CONFIG_PATA_SERVERWORKS is not set
 # CONFIG_PATA_SIL680 is not set
 # CONFIG_PATA_SIS is not set
@@ -2793,7 +2771,7 @@ CONFIG_MD_RAID0=y
 CONFIG_MD_RAID1=y
 CONFIG_MD_RAID10=y
 CONFIG_MD_RAID456=m
-CONFIG_BCACHE=m
+CONFIG_BCACHE=y
 # CONFIG_BCACHE_DEBUG is not set
 # CONFIG_BCACHE_ASYNC_REGISTRATION is not set
 CONFIG_BLK_DEV_DM_BUILTIN=y
@@ -2922,7 +2900,7 @@ CONFIG_ATM_DRIVERS=y
 # end of Distributed Switch Architecture drivers
 
 CONFIG_ETHERNET=y
-CONFIG_MDIO=m
+CONFIG_MDIO=y
 CONFIG_NET_VENDOR_3COM=y
 # CONFIG_VORTEX is not set
 # CONFIG_TYPHOON is not set
@@ -2950,7 +2928,7 @@ CONFIG_AMD_XGBE=y
 CONFIG_NET_XGENE=y
 # CONFIG_NET_XGENE_V2 is not set
 CONFIG_NET_VENDOR_AQUANTIA=y
-CONFIG_AQTION=m
+CONFIG_AQTION=y
 CONFIG_NET_VENDOR_ARC=y
 # CONFIG_EMAC_ROCKCHIP is not set
 CONFIG_NET_VENDOR_ASIX=y
@@ -2964,16 +2942,16 @@ CONFIG_ALX=m
 CONFIG_NET_VENDOR_BROADCOM=y
 # CONFIG_B44 is not set
 CONFIG_BCMGENET=y
-CONFIG_BNX2=m
+CONFIG_BNX2=y
 # CONFIG_CNIC is not set
-CONFIG_TIGON3=m
+CONFIG_TIGON3=y
 CONFIG_TIGON3_HWMON=y
-CONFIG_BNX2X=m
+CONFIG_BNX2X=y
 CONFIG_BNX2X_SRIOV=y
 CONFIG_BGMAC=y
 CONFIG_BGMAC_PLATFORM=y
 # CONFIG_SYSTEMPORT is not set
-CONFIG_BNXT=m
+CONFIG_BNXT=y
 CONFIG_BNXT_SRIOV=y
 CONFIG_BNXT_FLOWER_OFFLOAD=y
 # CONFIG_BNXT_DCB is not set
@@ -2989,7 +2967,7 @@ CONFIG_NET_VENDOR_CAVIUM=y
 # CONFIG_THUNDER_NIC_VF is not set
 # CONFIG_THUNDER_NIC_BGX is not set
 # CONFIG_THUNDER_NIC_RGX is not set
-CONFIG_CAVIUM_PTP=m
+CONFIG_CAVIUM_PTP=y
 # CONFIG_LIQUIDIO is not set
 # CONFIG_LIQUIDIO_VF is not set
 CONFIG_NET_VENDOR_CHELSIO=y
@@ -3002,7 +2980,7 @@ CONFIG_CHELSIO_T4VF=m
 CONFIG_CHELSIO_INLINE_CRYPTO=y
 CONFIG_CHELSIO_IPSEC_INLINE=m
 CONFIG_NET_VENDOR_CISCO=y
-CONFIG_ENIC=m
+CONFIG_ENIC=y
 CONFIG_NET_VENDOR_CORTINA=y
 # CONFIG_GEMINI_ETHERNET is not set
 CONFIG_NET_VENDOR_DAVICOM=y
@@ -3060,32 +3038,32 @@ CONFIG_NET_VENDOR_HUAWEI=y
 # CONFIG_HINIC3 is not set
 CONFIG_NET_VENDOR_I825XX=y
 CONFIG_NET_VENDOR_INTEL=y
-CONFIG_LIBETH=m
+CONFIG_LIBETH=y
 CONFIG_LIBETH_XDP=m
-CONFIG_LIBIE=m
+CONFIG_LIBIE=y
 CONFIG_LIBIE_ADMINQ=m
 CONFIG_LIBIE_FWLOG=m
-CONFIG_E100=m
-CONFIG_E1000=m
-CONFIG_E1000E=m
-CONFIG_IGB=m
+CONFIG_E100=y
+CONFIG_E1000=y
+CONFIG_E1000E=y
+CONFIG_IGB=y
 CONFIG_IGB_HWMON=y
-CONFIG_IGBVF=m
-CONFIG_IXGBE=m
+CONFIG_IGBVF=y
+CONFIG_IXGBE=y
 CONFIG_IXGBE_HWMON=y
 # CONFIG_IXGBE_DCB is not set
 CONFIG_IXGBE_IPSEC=y
-CONFIG_IXGBEVF=m
+CONFIG_IXGBEVF=y
 CONFIG_IXGBEVF_IPSEC=y
-CONFIG_I40E=m
+CONFIG_I40E=y
 # CONFIG_I40E_DCB is not set
-CONFIG_IAVF=m
-CONFIG_I40EVF=m
-CONFIG_ICE=m
+CONFIG_IAVF=y
+CONFIG_I40EVF=y
+CONFIG_ICE=y
 CONFIG_ICE_HWMON=y
 CONFIG_ICE_SWITCHDEV=y
 # CONFIG_FM10K is not set
-CONFIG_IGC=m
+CONFIG_IGC=y
 CONFIG_IDPF=m
 # CONFIG_IDPF_SINGLEQ is not set
 # CONFIG_JME is not set
@@ -3100,7 +3078,7 @@ CONFIG_MVPP2=y
 # CONFIG_MVPP2_PTP is not set
 # CONFIG_PXA168_ETH is not set
 # CONFIG_SKGE is not set
-CONFIG_SKY2=m
+CONFIG_SKY2=y
 # CONFIG_SKY2_DEBUG is not set
 # CONFIG_OCTEONTX2_AF is not set
 # CONFIG_OCTEONTX2_PF is not set
@@ -3184,15 +3162,15 @@ CONFIG_NET_VENDOR_PENSANDO=y
 # CONFIG_IONIC is not set
 CONFIG_NET_VENDOR_QLOGIC=y
 # CONFIG_QLA3XXX is not set
-CONFIG_QLCNIC=m
+CONFIG_QLCNIC=y
 CONFIG_QLCNIC_SRIOV=y
 CONFIG_QLCNIC_DCB=y
 CONFIG_QLCNIC_HWMON=y
 # CONFIG_NETXEN_NIC is not set
-CONFIG_QED=m
+CONFIG_QED=y
 CONFIG_QED_LL2=y
 CONFIG_QED_SRIOV=y
-CONFIG_QEDE=m
+CONFIG_QEDE=y
 CONFIG_QED_FCOE=y
 CONFIG_NET_VENDOR_BROCADE=y
 # CONFIG_BNA is not set
@@ -3205,12 +3183,12 @@ CONFIG_NET_VENDOR_RDC=y
 # CONFIG_R6040 is not set
 CONFIG_NET_VENDOR_REALTEK=y
 # CONFIG_8139CP is not set
-CONFIG_8139TOO=m
+CONFIG_8139TOO=y
 CONFIG_8139TOO_PIO=y
 # CONFIG_8139TOO_TUNE_TWISTER is not set
 # CONFIG_8139TOO_8129 is not set
 # CONFIG_8139_OLD_RX_RESET is not set
-CONFIG_R8169=m
+CONFIG_R8169=y
 # CONFIG_RTASE is not set
 CONFIG_NET_VENDOR_RENESAS=y
 # CONFIG_SH_ETH is not set
@@ -3247,24 +3225,24 @@ CONFIG_SMSC911X=y
 CONFIG_NET_VENDOR_SOCIONEXT=y
 # CONFIG_SNI_NETSEC is not set
 CONFIG_NET_VENDOR_STMICRO=y
-CONFIG_STMMAC_ETH=m
+CONFIG_STMMAC_ETH=y
 # CONFIG_STMMAC_SELFTESTS is not set
-CONFIG_STMMAC_PLATFORM=m
-CONFIG_DWMAC_DWC_QOS_ETH=m
-CONFIG_DWMAC_GENERIC=m
-CONFIG_DWMAC_IPQ806X=m
+CONFIG_STMMAC_PLATFORM=y
+CONFIG_DWMAC_DWC_QOS_ETH=y
+CONFIG_DWMAC_GENERIC=y
+CONFIG_DWMAC_IPQ806X=y
 # CONFIG_DWMAC_MEDIATEK is not set
-CONFIG_DWMAC_MESON=m
-CONFIG_DWMAC_QCOM_ETHQOS=m
+CONFIG_DWMAC_MESON=y
+CONFIG_DWMAC_QCOM_ETHQOS=y
 CONFIG_DWMAC_RENESAS_GBETH=m
-CONFIG_DWMAC_ROCKCHIP=m
-CONFIG_DWMAC_SUNXI=m
-CONFIG_DWMAC_SUN8I=m
+CONFIG_DWMAC_ROCKCHIP=y
+CONFIG_DWMAC_SUNXI=y
+CONFIG_DWMAC_SUN8I=y
 CONFIG_DWMAC_SUN55I=m
-CONFIG_DWMAC_IMX8=m
+CONFIG_DWMAC_IMX8=y
 # CONFIG_DWMAC_INTEL_PLAT is not set
-CONFIG_DWMAC_TEGRA=m
-CONFIG_STMMAC_PCI=m
+CONFIG_DWMAC_TEGRA=y
+CONFIG_STMMAC_PCI=y
 CONFIG_NET_VENDOR_SUN=y
 # CONFIG_HAPPYMEAL is not set
 # CONFIG_SUNGEM is not set
@@ -3370,7 +3348,7 @@ CONFIG_TERANETICS_PHY=y
 CONFIG_DP83822_PHY=y
 CONFIG_DP83TC811_PHY=y
 CONFIG_DP83848_PHY=y
-CONFIG_DP83867_PHY=m
+CONFIG_DP83867_PHY=y
 CONFIG_DP83869_PHY=y
 # CONFIG_DP83TD510_PHY is not set
 # CONFIG_DP83TG720_PHY is not set
@@ -3408,7 +3386,7 @@ CONFIG_MDIO_BUS_MUX_MMIOREG=y
 #
 # PCS device drivers
 #
-CONFIG_PCS_XPCS=m
+CONFIG_PCS_XPCS=y
 CONFIG_PCS_RZN1_MIIC=m
 # end of PCS device drivers
 
@@ -3419,7 +3397,7 @@ CONFIG_USB_NET_DRIVERS=y
 # CONFIG_USB_KAWETH is not set
 CONFIG_USB_PEGASUS=y
 CONFIG_USB_RTL8150=y
-CONFIG_USB_RTL8152=m
+CONFIG_USB_RTL8152=y
 CONFIG_USB_LAN78XX=y
 CONFIG_USB_USBNET=m
 CONFIG_USB_NET_AX8817X=m
@@ -3469,7 +3447,7 @@ CONFIG_USB_RTL8153_ECM=m
 # end of Wireless WAN
 
 CONFIG_XEN_NETDEV_FRONTEND=y
-CONFIG_VMXNET3=m
+CONFIG_VMXNET3=y
 # CONFIG_FUJITSU_ES is not set
 CONFIG_USB4_NET=m
 CONFIG_HYPERV_NET=y
@@ -3918,7 +3896,7 @@ CONFIG_I2C=y
 CONFIG_ACPI_I2C_OPREGION=y
 CONFIG_I2C_BOARDINFO=y
 CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_MUX=m
+CONFIG_I2C_MUX=y
 
 #
 # Multiplexer I2C Chip support
@@ -3937,7 +3915,7 @@ CONFIG_I2C_MUX=m
 
 CONFIG_I2C_HELPER_AUTO=y
 CONFIG_I2C_SMBUS=y
-CONFIG_I2C_ALGOBIT=m
+CONFIG_I2C_ALGOBIT=y
 
 #
 # I2C Hardware Bus support
@@ -3954,7 +3932,7 @@ CONFIG_I2C_CCGX_UCSI=y
 # CONFIG_I2C_AMD8111 is not set
 # CONFIG_I2C_AMD_MP2 is not set
 # CONFIG_I2C_HIX5HD2 is not set
-CONFIG_I2C_I801=m
+CONFIG_I2C_I801=y
 # CONFIG_I2C_ISCH is not set
 # CONFIG_I2C_PIIX4 is not set
 # CONFIG_I2C_NFORCE2 is not set
@@ -4071,7 +4049,7 @@ CONFIG_SPI_BCM_QSPI=y
 # CONFIG_SPI_HISI_KUNPENG is not set
 # CONFIG_SPI_HISI_SFC_V3XX is not set
 # CONFIG_SPI_NXP_FLEXSPI is not set
-# CONFIG_SPI_GPIO is not set
+CONFIG_SPI_GPIO=y
 # CONFIG_SPI_IMX is not set
 # CONFIG_SPI_FSL_SPI is not set
 # CONFIG_SPI_FSL_DSPI is not set
@@ -4185,7 +4163,7 @@ CONFIG_PINCTRL_AXP209=y
 CONFIG_PINCTRL_BM1880=y
 # CONFIG_PINCTRL_CY8C95X0 is not set
 CONFIG_PINCTRL_MAX77620=y
-# CONFIG_PINCTRL_MCP23S08 is not set
+CONFIG_PINCTRL_MCP23S08=y
 # CONFIG_PINCTRL_MICROCHIP_SGPIO is not set
 # CONFIG_PINCTRL_OCELOT is not set
 CONFIG_PINCTRL_RK805=y
@@ -4790,7 +4768,7 @@ CONFIG_AMLOGIC_THERMAL=y
 # Broadcom thermal drivers
 #
 CONFIG_BCM2711_THERMAL=y
-# CONFIG_BCM2835_THERMAL is not set
+CONFIG_BCM2835_THERMAL=y
 # CONFIG_BRCMSTB_THERMAL is not set
 CONFIG_BCM_NS_THERMAL=y
 CONFIG_BCM_SR_THERMAL=y
@@ -5883,7 +5861,7 @@ CONFIG_DRM_MESON_DW_MIPI_DSI=y
 # CONFIG_DRM_XEN_FRONTEND is not set
 # CONFIG_DRM_LIMA is not set
 CONFIG_DRM_PANFROST=m
-CONFIG_DRM_PANTHOR=m
+# CONFIG_DRM_PANTHOR is not set
 # CONFIG_DRM_TIDSS is not set
 # CONFIG_DRM_GUD is not set
 # CONFIG_DRM_ST7571_I2C is not set
@@ -6525,18 +6503,18 @@ CONFIG_HID_GENERIC=y
 #
 # Special HID drivers
 #
-CONFIG_HID_A4TECH=m
+CONFIG_HID_A4TECH=y
 # CONFIG_HID_ACCUTOUCH is not set
 # CONFIG_HID_ACRUX is not set
-CONFIG_HID_APPLE=m
+CONFIG_HID_APPLE=y
 # CONFIG_HID_APPLEIR is not set
 # CONFIG_HID_ASUS is not set
 # CONFIG_HID_AUREAL is not set
-CONFIG_HID_BELKIN=m
+CONFIG_HID_BELKIN=y
 # CONFIG_HID_BETOP_FF is not set
 # CONFIG_HID_BIGBEN_FF is not set
-CONFIG_HID_CHERRY=m
-CONFIG_HID_CHICONY=m
+CONFIG_HID_CHERRY=y
+CONFIG_HID_CHICONY=y
 # CONFIG_HID_CORSAIR is not set
 # CONFIG_HID_COUGAR is not set
 # CONFIG_HID_MACALLY is not set
@@ -6544,14 +6522,14 @@ CONFIG_HID_CHICONY=m
 # CONFIG_HID_CMEDIA is not set
 # CONFIG_HID_CP2112 is not set
 # CONFIG_HID_CREATIVE_SB0540 is not set
-CONFIG_HID_CYPRESS=m
+CONFIG_HID_CYPRESS=y
 # CONFIG_HID_DRAGONRISE is not set
 # CONFIG_HID_EMS_FF is not set
 # CONFIG_HID_ELAN is not set
 # CONFIG_HID_ELECOM is not set
 # CONFIG_HID_ELO is not set
 # CONFIG_HID_EVISION is not set
-CONFIG_HID_EZKEY=m
+CONFIG_HID_EZKEY=y
 # CONFIG_HID_FT260 is not set
 # CONFIG_HID_GEMBIRD is not set
 # CONFIG_HID_GFRM is not set
@@ -6569,17 +6547,17 @@ CONFIG_HID_EZKEY=m
 # CONFIG_HID_VIEWSONIC is not set
 # CONFIG_HID_VRC2 is not set
 # CONFIG_HID_XIAOMI is not set
-CONFIG_HID_GYRATION=m
+CONFIG_HID_GYRATION=y
 # CONFIG_HID_ICADE is not set
-CONFIG_HID_ITE=m
+CONFIG_HID_ITE=y
 # CONFIG_HID_JABRA is not set
 # CONFIG_HID_TWINHAN is not set
-CONFIG_HID_KENSINGTON=m
+CONFIG_HID_KENSINGTON=y
 # CONFIG_HID_LCPOWER is not set
 # CONFIG_HID_LED is not set
 # CONFIG_HID_LENOVO is not set
 # CONFIG_HID_LETSKETCH is not set
-CONFIG_HID_LOGITECH=m
+CONFIG_HID_LOGITECH=y
 # CONFIG_HID_LOGITECH_DJ is not set
 # CONFIG_HID_LOGITECH_HIDPP is not set
 CONFIG_LOGITECH_FF=y
@@ -6591,17 +6569,17 @@ CONFIG_LOGIWHEELS_FF=y
 # CONFIG_HID_MAYFLASH is not set
 # CONFIG_HID_MEGAWORLD_FF is not set
 CONFIG_HID_REDRAGON=y
-CONFIG_HID_MICROSOFT=m
-CONFIG_HID_MONTEREY=m
+CONFIG_HID_MICROSOFT=y
+CONFIG_HID_MONTEREY=y
 # CONFIG_HID_MULTITOUCH is not set
 # CONFIG_HID_NINTENDO is not set
 # CONFIG_HID_NTI is not set
 # CONFIG_HID_NTRIG is not set
 # CONFIG_HID_ORTEK is not set
-CONFIG_HID_PANTHERLORD=m
+CONFIG_HID_PANTHERLORD=y
 CONFIG_PANTHERLORD_FF=y
 # CONFIG_HID_PENMOUNT is not set
-CONFIG_HID_PETALYNX=m
+CONFIG_HID_PETALYNX=y
 # CONFIG_HID_PICOLCD is not set
 # CONFIG_HID_PLANTRONICS is not set
 # CONFIG_HID_PLAYSTATION is not set
@@ -6611,20 +6589,20 @@ CONFIG_HID_PETALYNX=m
 # CONFIG_HID_RETRODE is not set
 # CONFIG_HID_ROCCAT is not set
 # CONFIG_HID_SAITEK is not set
-CONFIG_HID_SAMSUNG=m
+CONFIG_HID_SAMSUNG=y
 # CONFIG_HID_SEMITEK is not set
 # CONFIG_HID_SIGMAMICRO is not set
 # CONFIG_HID_SONY is not set
 # CONFIG_HID_SPEEDLINK is not set
 # CONFIG_HID_STEAM is not set
 # CONFIG_HID_STEELSERIES is not set
-CONFIG_HID_SUNPLUS=m
+CONFIG_HID_SUNPLUS=y
 # CONFIG_HID_RMI is not set
 # CONFIG_HID_GREENASIA is not set
 # CONFIG_HID_HYPERV_MOUSE is not set
 # CONFIG_HID_SMARTJOYPLUS is not set
 # CONFIG_HID_TIVO is not set
-CONFIG_HID_TOPSEED=m
+CONFIG_HID_TOPSEED=y
 # CONFIG_HID_TOPRE is not set
 # CONFIG_HID_THINGM is not set
 # CONFIG_HID_THRUSTMASTER is not set
@@ -6832,13 +6810,13 @@ CONFIG_USB_SERIAL_GENERIC=y
 # CONFIG_USB_SERIAL_AIRCABLE is not set
 # CONFIG_USB_SERIAL_ARK3116 is not set
 # CONFIG_USB_SERIAL_BELKIN is not set
-CONFIG_USB_SERIAL_CH341=m
+CONFIG_USB_SERIAL_CH341=y
 # CONFIG_USB_SERIAL_WHITEHEAT is not set
 # CONFIG_USB_SERIAL_DIGI_ACCELEPORT is not set
-CONFIG_USB_SERIAL_CP210X=m
+CONFIG_USB_SERIAL_CP210X=y
 # CONFIG_USB_SERIAL_CYPRESS_M8 is not set
 # CONFIG_USB_SERIAL_EMPEG is not set
-CONFIG_USB_SERIAL_FTDI_SIO=m
+CONFIG_USB_SERIAL_FTDI_SIO=y
 # CONFIG_USB_SERIAL_VISOR is not set
 # CONFIG_USB_SERIAL_IPAQ is not set
 # CONFIG_USB_SERIAL_IR is not set
@@ -6859,7 +6837,7 @@ CONFIG_USB_SERIAL_FTDI_SIO=m
 # CONFIG_USB_SERIAL_MOS7840 is not set
 # CONFIG_USB_SERIAL_MXUPORT is not set
 # CONFIG_USB_SERIAL_NAVMAN is not set
-CONFIG_USB_SERIAL_PL2303=m
+CONFIG_USB_SERIAL_PL2303=y
 # CONFIG_USB_SERIAL_OTI6858 is not set
 # CONFIG_USB_SERIAL_QCAUX is not set
 # CONFIG_USB_SERIAL_QUALCOMM is not set
@@ -7070,24 +7048,24 @@ CONFIG_MMC_STM32_SDMMC=y
 CONFIG_MMC_SDHCI=y
 CONFIG_MMC_SDHCI_IO_ACCESSORS=y
 CONFIG_MMC_SDHCI_UHS2=m
-CONFIG_MMC_SDHCI_PCI=m
+CONFIG_MMC_SDHCI_PCI=y
 CONFIG_MMC_RICOH_MMC=y
-CONFIG_MMC_SDHCI_ACPI=m
-CONFIG_MMC_SDHCI_PLTFM=m
-CONFIG_MMC_SDHCI_OF_ARASAN=m
+CONFIG_MMC_SDHCI_ACPI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
 # CONFIG_MMC_SDHCI_OF_AT91 is not set
-CONFIG_MMC_SDHCI_OF_ESDHC=m
-CONFIG_MMC_SDHCI_OF_DWCMSHC=m
-CONFIG_MMC_SDHCI_CADENCE=m
+CONFIG_MMC_SDHCI_OF_ESDHC=y
+CONFIG_MMC_SDHCI_OF_DWCMSHC=y
+CONFIG_MMC_SDHCI_CADENCE=y
 # CONFIG_MMC_SDHCI_ESDHC_IMX is not set
-CONFIG_MMC_SDHCI_TEGRA=m
+CONFIG_MMC_SDHCI_TEGRA=y
 # CONFIG_MMC_SDHCI_PXAV3 is not set
-CONFIG_MMC_SDHCI_F_SDH30=m
+CONFIG_MMC_SDHCI_F_SDH30=y
 # CONFIG_MMC_SDHCI_MILBEAUT is not set
-CONFIG_MMC_SDHCI_IPROC=m
+CONFIG_MMC_SDHCI_IPROC=y
 CONFIG_MMC_MESON_GX=y
 # CONFIG_MMC_MESON_MX_SDIO is not set
-CONFIG_MMC_SDHCI_MSM=m
+CONFIG_MMC_SDHCI_MSM=y
 # CONFIG_MMC_MXC is not set
 # CONFIG_MMC_TIFM_SD is not set
 CONFIG_MMC_SPI=y
@@ -7117,8 +7095,8 @@ CONFIG_MMC_CQHCI=y
 # CONFIG_MMC_TOSHIBA_PCI is not set
 CONFIG_MMC_BCM2835=y
 # CONFIG_MMC_MTK is not set
-CONFIG_MMC_SDHCI_BRCMSTB=m
-CONFIG_MMC_SDHCI_XENON=m
+CONFIG_MMC_SDHCI_BRCMSTB=y
+CONFIG_MMC_SDHCI_XENON=y
 # CONFIG_MMC_SDHCI_AM654 is not set
 # CONFIG_MMC_OWL is not set
 # CONFIG_SCSI_UFSHCD is not set
@@ -7222,10 +7200,8 @@ CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
 #
 # CONFIG_ACCESSIBILITY is not set
 CONFIG_INFINIBAND=y
-CONFIG_INFINIBAND_USER_MAD=m
-CONFIG_INFINIBAND_USER_ACCESS=m
-CONFIG_INFINIBAND_USER_MEM=y
-CONFIG_INFINIBAND_ON_DEMAND_PAGING=y
+# CONFIG_INFINIBAND_USER_MAD is not set
+# CONFIG_INFINIBAND_USER_ACCESS is not set
 CONFIG_INFINIBAND_ADDR_TRANS=y
 CONFIG_INFINIBAND_ADDR_TRANS_CONFIGFS=y
 CONFIG_INFINIBAND_VIRT_DMA=y
@@ -7532,17 +7508,17 @@ CONFIG_VMGENID=y
 # CONFIG_ARM_CCA_GUEST is not set
 CONFIG_VIRTIO_ANCHOR=y
 CONFIG_VIRTIO=y
-CONFIG_VIRTIO_PCI_LIB=m
-CONFIG_VIRTIO_PCI_LIB_LEGACY=m
+CONFIG_VIRTIO_PCI_LIB=y
+CONFIG_VIRTIO_PCI_LIB_LEGACY=y
 CONFIG_VIRTIO_MENU=y
-CONFIG_VIRTIO_PCI=m
+CONFIG_VIRTIO_PCI=y
 CONFIG_VIRTIO_PCI_LEGACY=y
 CONFIG_VIRTIO_VDPA=m
 CONFIG_VIRTIO_PMEM=y
-CONFIG_VIRTIO_BALLOON=m
+CONFIG_VIRTIO_BALLOON=y
 # CONFIG_VIRTIO_MEM is not set
-CONFIG_VIRTIO_INPUT=m
-CONFIG_VIRTIO_MMIO=m
+CONFIG_VIRTIO_INPUT=y
+CONFIG_VIRTIO_MMIO=y
 CONFIG_VIRTIO_MMIO_CMDLINE_DEVICES=y
 CONFIG_VIRTIO_DMA_SHARED_BUFFER=y
 # CONFIG_VIRTIO_DEBUG is not set
@@ -9069,7 +9045,7 @@ CONFIG_PWM=y
 CONFIG_PWM_BCM2835=y
 CONFIG_PWM_BCM_IPROC=y
 # CONFIG_PWM_BERLIN is not set
-# CONFIG_PWM_BRCMSTB is not set
+CONFIG_PWM_BRCMSTB=y
 # CONFIG_PWM_CLK is not set
 # CONFIG_PWM_DWC is not set
 # CONFIG_PWM_FSL_FTM is not set
@@ -9776,8 +9752,29 @@ CONFIG_INTEGRITY_TRUSTED_KEYRING=y
 CONFIG_INTEGRITY_PLATFORM_KEYRING=y
 CONFIG_LOAD_UEFI_KEYS=y
 CONFIG_INTEGRITY_AUDIT=y
-# CONFIG_IMA is not set
-# CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT is not set
+CONFIG_IMA=y
+# CONFIG_IMA_KEXEC is not set
+CONFIG_IMA_MEASURE_PCR_IDX=10
+CONFIG_IMA_LSM_RULES=y
+CONFIG_IMA_NG_TEMPLATE=y
+# CONFIG_IMA_SIG_TEMPLATE is not set
+CONFIG_IMA_DEFAULT_TEMPLATE="ima-ng"
+# CONFIG_IMA_DEFAULT_HASH_SHA1 is not set
+# CONFIG_IMA_DEFAULT_HASH_SHA256 is not set
+CONFIG_IMA_DEFAULT_HASH_SHA512=y
+CONFIG_IMA_DEFAULT_HASH="sha512"
+CONFIG_IMA_WRITE_POLICY=y
+CONFIG_IMA_READ_POLICY=y
+CONFIG_IMA_APPRAISE=y
+CONFIG_IMA_ARCH_POLICY=y
+# CONFIG_IMA_APPRAISE_BUILD_POLICY is not set
+CONFIG_IMA_APPRAISE_BOOTPARAM=y
+# CONFIG_IMA_APPRAISE_MODSIG is not set
+# CONFIG_IMA_BLACKLIST_KEYRING is not set
+# CONFIG_IMA_LOAD_X509 is not set
+CONFIG_IMA_MEASURE_ASYMMETRIC_KEYS=y
+CONFIG_IMA_QUEUE_EARLY_BOOT_KEYS=y
+CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT=y
 # CONFIG_EVM is not set
 # CONFIG_DEFAULT_SECURITY_SELINUX is not set
 # CONFIG_DEFAULT_SECURITY_APPARMOR is not set
@@ -9975,7 +9972,7 @@ CONFIG_CRYPTO_LZO=y
 # CONFIG_CRYPTO_842 is not set
 # CONFIG_CRYPTO_LZ4 is not set
 # CONFIG_CRYPTO_LZ4HC is not set
-CONFIG_CRYPTO_ZSTD=y
+CONFIG_CRYPTO_ZSTD=m
 # end of Compression
 
 #
-- 
2.51.1

